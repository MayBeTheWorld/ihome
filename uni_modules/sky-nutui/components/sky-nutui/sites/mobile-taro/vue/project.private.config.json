{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "feedback/pages/notify/index",
          "pathName": "feedback/pages/notify/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "business/pages/signature/index",
          "pathName": "business/pages/signature/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/collapse/index",
          "pathName": "exhibition/pages/collapse/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "NutUI-Taro"
}