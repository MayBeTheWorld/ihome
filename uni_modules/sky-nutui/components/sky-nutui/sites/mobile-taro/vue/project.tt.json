{
  "miniprogramRoot": "./",
  "projectname": "@nutui/nutui-taro-mobile",
  "appid": "touristappid",
  "setting": {
    "es6": false,
    "minified": false
  }
}
