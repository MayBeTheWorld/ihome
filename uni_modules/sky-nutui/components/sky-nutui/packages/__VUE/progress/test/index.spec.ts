import { config, mount } from '@vue/test-utils';
import { nextTick } from 'vue';
import NutIcon from '../../icon/index.vue';
import Progress from '../index.vue';

beforeAll(() => {
  config.global.components = {
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

test('should render progress when use props', async () => {
  const wrapper = mount(Progress, {
    props: {
      percentage: 100
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});
test('should render different height and color when use color height props', async () => {
  const wrapper = mount(Progress, {
    props: {
      percentage: 50,
      strokeColor: 'blue',
      strokeWidth: '20',
      textColor: 'red'
    }
  });
  await nextTick();
  let inner = wrapper.find<HTMLElement>('.nut-progress-inner');
  expect(inner.element.style.background).toEqual('blue');
  let text = wrapper.find<HTMLElement>('.nut-progress-text');
  let span = wrapper.find<HTMLElement>('.nut-progress-text span');
  expect(span.element.style.color).toEqual('red');
  expect(text.element.style.lineHeight).toEqual('20px');
});

test('should hide percentage when use showText props', () => {
  const wrapper = mount(Progress, {
    props: {
      percentage: 50,
      showText: false
    }
  });
  let text = wrapper.find<HTMLElement>('.nut-progress-text');
  expect(text.exists()).toBe(false);
});

test('should render inside percentage when use textInside props', () => {
  const wrapper = mount(Progress, {
    props: {
      percentage: 50,
      textInside: true
    }
  });
  let text = wrapper.find<HTMLElement>('.nut-progress-insidetext');
  expect(text.exists()).toBe(true);
});

test('should render custom size when use size props', () => {
  const wrapper = mount(Progress, {
    props: {
      percentage: 50,
      size: 'large'
    }
  });
  let text = wrapper.find<HTMLElement>('.nut-progress-large');
  expect(text.exists()).toBe(true);
});
