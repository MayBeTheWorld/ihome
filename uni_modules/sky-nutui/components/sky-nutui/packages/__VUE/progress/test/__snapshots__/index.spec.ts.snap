// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render progress when use props 1`] = `
"<div class=\\"nut-progress\\">
  <div class=\\"nut-progress-outer nut-progress-outer-part nut-progress-base\\">
    <div class=\\"nut-progress-inner\\" style=\\"width: 100%;\\">
      <!--v-if-->
      <!--v-if-->
    </div>
  </div>
  <div class=\\"nut-progress-text\\" style=\\"line-height: px;\\"><span>100%</span></div>
</div>"
`;
